Modules/overlapped.c

```diff
@@ -602,6 +612,21 @@ Overlapped_dealloc(OverlappedObject *self)
     case TYPE_ACCEPT:
         Py_CLEAR(self->allocated_buffer);
         break;
+    case TYPE_READ_FROM:
+        // An initial call to WSARecvFrom will only allocate the buffer.
+        // The result tuple of (message, (address, port)) is only
+        // allocated _after_ a message has been received.
+        if(self->read_from.result) {
+            // We've received a message, free the result tuple which will
+            // also free the message buffer.
```

Comment:
This seems like an error-prone pattern, and it's just as easy to Py_INCREF when adding the buffer into the tuple and Py_CLEAR here (rather than using borrowed references).